indexing-service/src/main/java/org/apache/druid/indexing/common/task/KillUnusedSegmentsTask.java

```diff
@@ -129,8 +129,12 @@ public TaskStatus runTask(TaskToolbox toolbox) throws Exception
 
     // Kill segments
     toolbox.getTaskActionClient().submit(new SegmentNukeAction(new HashSet<>(unusedSegments)));
-    for (DataSegment segment : unusedSegments) {
-      toolbox.getDataSegmentKiller().kill(segment);
+    if (getContextValue("batchDelete", false)) {
```

Comment:
Ah i was thinking the otherway around. have it be opt in at first, then opt out. Just to make it be for the more safety conscious users to have a chance to migrate to it. Never got around to imagining of removing it entirely. Im just hung up the whole feature flag and how no other extension can take advantage of it. So it's more of a best effort feature flag.

I would need to look into auto-kill and how that works

Reply:
IMO enabling this new feature by default is best, as it's better and the risk isn't super high.

I'm not even sure we need a feature flag at all. We should be able to get the risk low enough with appropriate testing that a flag isn't needed.

If we're gonna have one, then yeah, it needs to be settable for people that use autokill.

Reply:
As to testing: one thing to think about is whether this uses any new S3 APIs and therefore may require new IAM permissions. If it would require new IAM permissions then we definitely need a feature flag and we need to call it out in the docs and release notes. Or, better, adjust the design to not use new APIs and not need new IAM permissions.

Reply:
If we decide to go the "not require new APIs" route — which IMO is best — then we should have a note in the interface that says that the multi-arg form of `kill` must not require additional permissions on the deep storage beyond what the single-arg form of `kill` requires. Otherwise, when a multi-arg impl is contributed for another deep storage, it could break people in production that don't have the right set of permissions.

Reply:
@gianm I decided to enable this by default and not have a feature flag for it at all. The api that is used to delete multiple keys from a bucket is already used elsewhere in the extension. 

I added that verbiage for kill not require additional permissions